src/ZODB/tests/util.py

```diff
@@ -338,3 +339,41 @@ def testSetUp(self):
     def testTearDown(self):
         self.time_manager.close()
         reset_monotonic_time()
+
+
+def with_high_concurrency(f):
+    """
+    with_high_concurrency decorates f to run with high frequency of thread context switches.
+
+    It is useful for tests that try to probabilistically reproduce race
+    condition scenarios.
+    """
+    @functools.wraps(f)
+    def _(*argv, **kw):
+        if six.PY3:
```

Comment:
Can you move to an upper level so we can have:
```py
previous = sys.getswitchinterval()
try:
    sys.setswitchinterval(5e-6)
    return f(*argv, **kw)
finally:
    sys.setswitchinterval(previous)
```
and
```py
previous = sys.getcheckinterval()
try:
    sys.setcheckinterval(100)
    return f(*argv, **kw)
finally:
    sys.setcheckinterval(previous)
```

Reply:
It makes the code less clear to me. I prefer to keep my original version.

Reply:
One could argue about the duplication of the try/return/finally lines, even if for me it's so small that I'd call it "overfactoring". But less clear? There's nothing more pythonic than the following pattern:
```
save state
try:
    mess up (and ideally, do it after the try - see PEP 419)
    ...
finally:
    restore
```
No need to "name" the restore part by moving it a function.

When we drop support for Python 2, that's anyway how it would be rewritten.

Reply:
> One could argue about the duplication of the try/return/finally lines

Yes, that duplication of f call sites makes it less clear in my view.

> When we drop support for Python 2, that's anyway how it would be rewritten.

So let's rewrite it that way _if_ | _when_ Python 2 support is dropped. And until it is not dropped, let's keep the code clearly structured.